src/main/java/io/openliberty/tools/intellij/util/LibertyMavenUtil.java

```diff
@@ -289,6 +296,20 @@ private static String getCustomMavenPath(Project project, String customMavenHome
         }
     }
 
+    /**
+     * This method checks if the provided 'MavenHomeType' is an instance of 'StaticResolvedMavenHomeType'.
+     * If it is, it returns the casted object. Otherwise, it uses a fallback method (staticOrBundled) to return default 'BundledMaven3.INSTANCE'
+     * @param customMavenHome the custom MavenHomeType to be resolved
+     * @return the resolved StaticResolvedMavenHomeType
+     */
+    private static StaticResolvedMavenHomeType getResolvedMavenHomeType(MavenHomeType customMavenHome) {
+        if (customMavenHome instanceof StaticResolvedMavenHomeType) {
+            return (StaticResolvedMavenHomeType) customMavenHome;
+        }
+        // If 'customMavenHome' is not an instance of 'StaticResolvedMavenHomeType', return default 'BundledMaven3.INSTANCE'.
+        return staticOrBundled(customMavenHome);
```

Comment:
I have a couple questions around using the `staticOrBundled()` method here.
1. Is this method intended to be part of the public API for the jetbrains maven library? Or is it an internal method?
2. If the behaviour of `staticOrBundled()` is to attempt to cast the input param to `StaticResolvedMavenHomeType` or return `BundledMaven3.INSTANCE`, could we not just do that ourselves here? We already check for the instance type with the `if` statement. If that is `false`, I think that means that we can assume `BundledMaven3.INSTANCE` will be returned from `staticOrBundled()`... or am I missing something?

To summarize, if the `staticOrBundled()` method is an internal method of the jetbrains maven library, we probably should not use it and just mock its behaviour in the `getResolvedMavenHomeType()` method. If it is intended to be part of the publicly available API, then I think we can just use the `staticOrBundled()` method directly from `getCustomMavenPath()` and cut out the "middle man" method `getResolvedMavenHomeType()`.

Reply:
1. Yes @TrevCraw , `staticOrBundled()` method intended to be part of the public API for the jetbrains maven library and not an internal method
<img width="1058" alt="Screenshot 2024-06-25 at 1 54 59 PM" src="https://github.com/OpenLiberty/liberty-tools-intellij/assets/133869023/905de854-0f72-457e-a0ec-546a64ce512f">

2. I modified the code to use the `staticOrBundled()` method directly from `getCustomMavenPath(`) and removed  method`(getResolvedMavenHomeType())` that I created.

Reply:
Hi @vaisakhkannan , the changes look good. Did you re-test with these new changes?

Reply:
Yes @TrevCraw , I did testing in all 3 maven settings and its working fine